src/groupByKey.ts

```diff
@@ -2,6 +2,21 @@ type GroupsMap<T> = {
   [key: string]: T[];
 };
 
-export function groupByKey(items, key) {
+export function groupByKey<
+  T, R extends keyof T
+>(
+  items: T[], key: R,
```

Comment:
You need only 1 generic type here
```suggestion
export function groupByKey<T>(
  items: T[], key: keyof T,
```